src/daft-core/src/kernels/search_sorted.rs

```diff
@@ -142,6 +142,77 @@ fn search_sorted_utf_array<O: Offset>(
     PrimitiveArray::<u64>::new(DataType::UInt64, results.into(), None)
 }
 
+fn search_sorted_boolean_array(
+    sorted_array: &BooleanArray,
+    keys: &BooleanArray,
+    input_reversed: bool,
+) -> PrimitiveArray<u64> {
+    let array_size = sorted_array.len();
+    let mut left = 0_usize;
+    let mut right = array_size;
+
+    // For boolean arrays, we know there can only be three possible values: true, false, and null.s
+    // We can pre-compute the results for these three values and then reuse them to compute the results for the keys.
+    let pre_computed_keys = &[Some(true), Some(false), None];
+    let mut pre_computed_results: HashMap<Option<bool>, u64> = HashMap::new();
```

Comment:
Hashmaps are expensive, you should be able to have 3 variables. Maybe just use a `[usize]`?